kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/dhcp/DhcpClientLeaseBlock.java

```diff
@@ -0,0 +1,268 @@
+    public IPAddress getFixedAddress() {
+        return this.fixedAddress;
+    }
+
+    /**
+     * Reports subnet mask
+     * 
+     * @return subnet mask
+     */
+    public IPAddress getSubnetMask() {
+        return this.subnetMask;
+    }
+
+    /**
+     * Reports list of routers
+     * 
+     * @return list of routers
+     */
+    public List<IPAddress> getRouters() {
+        return this.routers;
+    }
+
+    /**
+     * Reports DHCP lease time
+     * 
+     * @return DHCP lease time
+     */
+    public int getDhcpLeaseTime() {
+        return this.dhcpLeaseTime;
+    }
+
+    /**
+     * Reports DHCP message type
+     * 
+     * @return DHCP message type
+     */
+    public int getDhcpMessageType() {
+        return this.dhcpMessageType;
+    }
+
+    /**
+     * Reports list of DNS servers
+     * 
+     * @return list of DNS servers
+     */
+    public List<IPAddress> getDnsServers() {
+        return this.dnsServers;
+    }
+
+    /**
+     * Reports DHCP server
+     * 
+     * @return DHCP server
+     */
+    public IPAddress getDhcpServer() {
+        return this.dhcpServer;
+    }
+
+    /**
+     * Reports DHCP renewal time
+     * 
+     * @return DHCP renewal time
+     */
+    public int getDhcpRenewalTime() {
+        return this.dhcpRenewalTime;
+    }
+
+    /**
+     * Reports DHCP rebinding time
+     * 
+     * @return DHCP rebinding time
+     */
+    public int getDhcpRebindingTime() {
+        return this.dhcpRebindingTime;
+    }
+
+    /**
+     * Reports domain name
+     * 
+     * @return domain name
+     */
+    public String getDomainName() {
+        return this.domainName;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("Dhclient Lease Block: [");
+        sb.append(INTERFACE_PARAM_NAME).append('=').append(this.iface);
+        sb.append("; ").append(FIXED_ADDRESS_PARAM_NAME).append('=').append(this.fixedAddress);
+        sb.append("; ").append(SUBNET_MASK_OPTION_NAME).append('=').append(this.subnetMask);
+        sb.append("; ").append(ROUTERS_OPTION_NAME).append('=').append(this.routers);
+        sb.append("; ").append(DHCP_LEASE_TIME_OPTION_NAME).append('=').append(this.dhcpLeaseTime);
+        sb.append("; ").append(DHCP_MESSAGE_TYPE_OPTION_NAME).append('=').append(this.dhcpMessageType);
+        sb.append("; ").append(DOMAIN_NANE_SERVERS_OPTION_NAME).append('=').append(this.dnsServers);
+        sb.append("; ").append(DHCP_SERVER_IDENTIFIER_OPTION_NAME).append('=').append(this.dhcpServer);
+        sb.append("; ").append(DHCP_RENEWAL_TIME_OPTION_NAME).append('=').append(this.dhcpRenewalTime);
+        sb.append("; ").append(DHCP_REBINDING_TIME_OPTION_NAME).append('=').append(this.dhcpRebindingTime);
+        sb.append("; ").append(DOMAIN_NAME_OPTION_NAME).append('=').append(this.domainName);
+        sb.append(']');
+        return sb.toString();
+    }
+
+    private int parseInteger(String line, String name) {
```

Comment:
It would be better to let the caller handle Exceptions

Reply:
done